### DISCUSSION TITLE
Ler digital com JS

### DISCUSSION THREAD

**Discussion by albertopida** [frontendbr/forum#1068] [Perguntas e Respostas]
Bom dia,

Gostaria de saber se alguém ja conseguiu usar um leitor de digital com JS no browser?

Obrigado

**Comment by felipefialho** [comment]
Tem o [fingerprintjs](https://github.com/Valve/fingerprintjs2)

**Comment by kivervinicius** [comment]
@LFeh acredito que ele quis dizer um leitor de digitais mesmo, rs, há uns 10 anos atrás eu fiz um projeto com o Hamster II da Nitgen, na época havia um SDK que havia JS porém utilizava ActiveX porém como sabem já foi descontinuado nem é recomendado, seria interessante verificar com um fabricante e verificar a disponibilidade.

**Comment by felipefialho** [comment]
Achei que o fingerprintjs servisse pra isso também haha

**Comment by munizart** [comment]
Seria necessario configurar um servidor na maquina com o leitor de digital pro browser interagir (via http ou ws).
Outra forma é criar um driver que transforme o leitor de digital num gamepad e fazer um protocolo meio gambiarra em cima disso.
Outra alternativa pra um futuro breve é o WebAuthn API, que segundo a [especificação](https://www.w3.org/TR/webauthn/) permitiria autenticação por biometria.

UPDATE:
Na quarta-feira, dia 09, sai a versão 60 do firefox já com suporte pra WebAuthn. Pra quem usa Chrome o suporte deve chegar até o fim do mês.

**Comment by marconimotta** [comment]
@albertopida desculpa perguntar, mas você conseguiu implementar? Estou tentando fazer o mesmo e não acho muito conteúdo para auxiliar.
